### ISSUE TITLE
Display Carbon Tax Exemption ID when Operation is Approved

### ISSUE THREAD

**Issue by andrea-williams** [bcgov/cas-registration#273]
#### Description:

This captures the final piece of the carbon tax exemption workflow: when an Operation's carbon tax exemption status is Approved, what happens next?

Number format proposed by business: yy-xxxx (two digit year followed by consecutive numbering). E.g. 24-0001

If the submission in rejected, show rejection page; no Exemption ID is generated. 

Wireframe: https://www.figma.com/file/LsDmLDCdnJqI8UahyqvOD8/OBPS-App-Designs?type=design&node-id=340%3A383&mode=design&t=5ZQBmcc61TmYu2p1-1

#### Acceptance Criteria:

Given I am a CAS internal staff member
When I approve an Operation's application
Then a carbon tax exemption certificate number should be generated, stored in the database, and displayed on screen

Given I am an IO user with admin or reporter role for a certain Operation
When my Operation has been approved for carbon tax exemption
Then I can view my operation's exemption number 

#### Development Checklist:

- [x] each approved Operation's exemption ID number should be unique ~~- we'll need to test this thoroughly. (Andrea's opinion: ideally, exemption ID number generated by database rather than Python backend. This will ensure uniqueness and sequential ordering without race conditions. Will also greatly simplify testing because we can trust Postgres's logic much more than our own custom logic.)~~
- [x] best approach: use Django's `unique` prop on the field to ensure uniqueness (and to greatly simplify testing).
- [ ] Meets the DOD

**Definition of Ready** (Note: If any of these points are not applicable, mark N/A)

- [x] User story is included
- [ ] User role and type are identified
- [x] Acceptance criteria are included
- [x] Wireframes are included (if required)
- [x] Design / Solution is accepted by Product Owner
- [ ] Dependencies are identified (technical, business, regulatory/policy)
- [ ] Story has been estimated (under 13 pts)

·**Definition of Done** (Note: If any of these points are not applicable, mark N/A)

- [ ] Acceptance criteria are tested by the CI pipeline
- [ ] UI meets accessibility requirements
- [ ] Configuration changes are documented, documentation and designs are updated
- [ ] Passes code peer-review
- [ ] Passes QA of Acceptance Criteria with verification in Dev and Test
- [ ] Ticket is ready to be merged to main branch
- [ ] Can be demoed in Sprint Review
- [ ] Bugs or future work cards are identified and created
- [ ] Reviewed and approved by Product Owner

#### Notes:

-

**Comment by Dianadec** [comment]
This could be estimated while we wait to hear back about the exact numbering convention, but it cannot be worked on until we have that number.

**Comment by hannavovk** [comment]
Number format proposed by business are: yy-xxxx (two digit year followed by consecutive numbering). E.g. 24-0001.
Dev's, please let me know if you see any issues with this format.

**Comment by hannavovk** [comment]
The page can be built with draft text as currently shown in wireframes and may need to be updated later.

**Comment by Dianadec** [comment]
Design just needs a few updates and then the wireframe link will be added to this ticket.

**Comment by Dianadec** [comment]
Hey team! Please [add your planning poker estimate](https://app.zenhub.com/workspaces/Team-Moose-Sprint-Board-60d216175a2689000ec774d0/issues/bcgov/cas-registration/273?planning-poker) with Zenhub @andrea-williams @BCerki @dleard @JoshLarouche @marcellmueller @Sepehr-Sobhani @shon-button

**Comment by Sepehr-Sobhani** [comment]
@andrea-williams I believe that assigning a unique property to the data field would ensure its uniqueness. Therefore, I strongly recommend using the built-in functionalities of Django instead of relying on raw SQL queries. This way, we can maintain a cleaner workflow during development and testing.

**Comment by andrea-williams** [comment]
thanks @Sepehr-Sobhani - I've updated the Development Checklist in this ticket to incorporate your suggestion

**Comment by Dianadec** [comment]
@hannavovk noted that the name will change - we will create a separate TINY card to address that once the name is approved.

**Comment by hannavovk** [comment]
The ID name has now been finalized: "Exemption ID" is now "BC OBPS Regulated Operation ID" (or BORO ID).
If it makes sense to change the name as part of this ticket, feel free to do that. Otherwise we can make that a separate task. 
Note that other text on this page is yet to be finalized. 
cc @Sepehr-Sobhani @andrea-williams @dleard

**Comment by Sepehr-Sobhani** [comment]
I have created this task ticket: https://app.zenhub.com/workspaces/team-moose-sprint-board-60d216175a2689000ec774d0/issues/gh/bcgov/cas-registration/427
Please feel free to refine it.
@dleard Have I included all the required data fields?

**Comment by dleard** [comment]
perhaps a status column for valid/revoked? Other than that this looks good, we could also ask Jairo if there's any other metadata about the exemption ID that CAS thinks would be helpful

**Comment by Sepehr-Sobhani** [comment]
Will be addressed by ticket: #427
